--- user_form.php_default	2009-04-13 18:51:35.000000000 +0200
+++ user_form.php	2009-04-13 18:49:55.000000000 +0200
@@ -54,6 +54,7 @@
 			</tr>
 		</table>
 	</fieldset><br />
+    <?php if (TZN_USER_PASS_MODE != 5) { ?>
 	<fieldset>
 		<legend><?php echo $langUser['account']; ?></legend>
 		<p><?php echo $langUser['account_legend']; ?></p>
@@ -100,3 +101,4 @@
 			}
 		?>
 	</fieldset>
+    <?php } ?>
